src/Apps/Order/redirects.tsx

```diff
@@ -186,3 +205,39 @@ export const redirects: RedirectRecord<{
     },
   ],
 }
+
+graphql`
+  fragment redirects_order on Order {
+    id
+    mode
+    state
+    ... on OfferOrder {
+      myLastOffer {
+        id
+        createdAt
+      }
+      lastOffer {
+        id
+        createdAt
+      }
+    }
+    requestedFulfillment {
+      __typename
+    }
+    lineItems {
+      edges {
+        node {
+          artwork {
+            id
+          }
+        }
+      }
+    }
+    creditCard {
+      id
+    }
```

Comment:
Might be worth it to consider adding `hasFoo` fields to MP for all these fields that have a dummy selection (`id`) only to test for its existence.

Reply:
I agree dummy selections are unfortunate. Could `has*` fields be added automatically for any nullable non-primitive type?

Reply:
Interesting thought. I guess we could have our own factory that would emit the object and the boolean fields, but that’s not something we have today.

Reply:
Until then, though, you may want to consider selecting the `__id` field instead, if the type has that, because Relay will already implicitly select that anyways and so you won’t be pulling more data over the network.